src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -14680,5 +14686,124 @@ enum E { }
                 }
             }
         }
+
+        [Theory, CombinatorialData]
+        public void BetterConversionTarget(bool nullable1, bool nullable2)
+        {
+            // Given two types T1 and T2, T1 is a better conversion target than T2 if one of the following holds:
+            // 1. An implicit conversion from T1 to T2 exists and no implicit conversion from T2 to T1 exists
+            // ...
+            // 3. T1 is S1 or S1? where S1 is a signed integral type, and T2 is S2 or S2? where S2 is an unsigned integral type.
+
+            string s1Nullable = nullable1 ? "?" : "";
+            string s2Nullable = nullable2 ? "?" : "";
+
+            string source = $$"""
+using static System.Console;
+
+C.M1(0);
+C.M2(0);
+C.M3(0);
+C.M4(0);
+C.M5(0);
+C.M6(0);
+C.M7(0);
+C.M8(0);
+
+public class C
+{
+    public static void M1(sbyte{{s1Nullable}} x) { Write("M1 ");  }
+    public static void M1(nuint{{s2Nullable}} x) { }
+
+    public static void M2(short{{s1Nullable}} x) { Write("M2 "); }
+    public static void M2(nuint{{s2Nullable}} x) { }
+
+    public static void M3(int{{s1Nullable}} x) { Write("M3 "); }
+    public static void M3(nuint{{s2Nullable}} x) { }
+
+    public static void M4(long{{s1Nullable}} x) { Write("M4 "); }
+    public static void M4(nuint{{s2Nullable}} x) { }
+
+    public static void M5(nint{{s1Nullable}} x) { Write("M5(nint) "); }
+    public static void M5(ushort{{s2Nullable}} x) { Write("M5(ushort) ");  }
+
+    public static void M6(nint{{s1Nullable}} x) { Write("M6 "); }
+    public static void M6(uint{{s2Nullable}} x) { }
+
+    public static void M7(nint{{s1Nullable}} x) { Write("M7 "); }
+    public static void M7(ulong{{s2Nullable}} x) { }
+
+    public static void M8(nint{{s1Nullable}} x) { Write("M8"); }
+    public static void M8(nuint{{s2Nullable}} x) { }
+}
```

Comment:
Should this set include the following pair? Similar question for test below.
```csharp
static void M(byte x) { }
static void M(nint x) { }
```

Reply:
I'll add it, but that case is different because there's a clear better type (one implicitly converts to the other but not vice-versa), so we don't hit the better conversion target rule 3.

Reply:
It looks like the version of the test in NumericIntPtrTests was changed, but not this version.